send/m365_cloud

```diff
@@ -466,21 +483,21 @@ def remove_groups(our_groups, their_groups):
 	for their_group in their_groups:
 		if not their_group.get(PERUN_MANAGED_EXTENSION_NAME):
 			continue
-		our_group = next(filter(lambda g: g[groupname_attr_name] == their_group["displayName"], our_groups), None)
+		our_group = next(filter(lambda g: g[resource_id_attr_name] == their_group[PERUN_ID_EXTENSION_NAME], our_groups), None)
 		if our_group is None:
 			remove_group(their_group)
 
 
 def create_update_restore_groups(our_groups, removed_groups, their_groups):
 	""" Checks which groups are missing in m365 and creates them, updates members of existing groups """
 	for our_group in our_groups:
-		their_group = next(filter(lambda g: g["displayName"] == our_group[groupname_attr_name], their_groups), None)
+		their_group = next(filter(lambda g: g.get(PERUN_ID_EXTENSION_NAME) == our_group[resource_id_attr_name], their_groups), None)
 		if their_group is None:
-			their_removed_group = next(filter(lambda g: g["displayName"] == our_group[groupname_attr_name], removed_groups), None)
-			if (their_removed_group and their_removed_group.get(PERUN_MANAGED_EXTENSION_NAME)
-				and their_removed_group.get(PERUN_ID_EXTENSION_NAME, -1) == our_group[resource_id_attr_name]):
+			their_removed_group = next(filter(lambda g: g.get(PERUN_MANAGED_EXTENSION_NAME, False)
+				and g.get(PERUN_ID_EXTENSION_NAME) == our_group[resource_id_attr_name], removed_groups), None)
```

Comment:
```suggestion
				and g.get(PERUN_ID_EXTENSION_NAME, -1) == our_group[resource_id_attr_name], removed_groups), None)
```
Not sure this is necessary but in case you missed it